hal_st/stm32fxxx/AnalogToDigitalPinStm.hpp

```diff
@@ -28,12 +44,13 @@ namespace hal
         : public AnalogToDigitalPinImplBase<uint16_t>
     {
     public:
-        explicit AnalogToDigitalInternalTemperatureStm(AdcStm& adc);
+        explicit AnalogToDigitalInternalTemperatureStm(AdcStm& adc, detail::AdcStmChannelConfig adcStmChannelConfig);
```

Comment:
```suggestion
        using Config = detail::AdcStmChannelConfig;

        AnalogToDigitalInternalTemperatureStm(AdcStm& adc, const Config& config = Config());
```

- `detail` is usually hidden from the user
- most users will use default value for this setting